Compilation error: math_function_generated_im2col.cu.o

```
/home/luotao02/Paddle/paddle/fluid/framework/dim.h(161): error: Within a __device__/__global__ function, only __shared__ variables may be marked "static"

/home/luotao02/Paddle/paddle/fluid/framework/dim.h(187): error: Within a __device__/__global__ function, only __shared__ variables may be marked "static"

2 errors detected in the compilation of "/home/luotao02/.tmp/tmpxft_00003712_00000000-7_im2col.cpp1.ii".
CMake Error at math_function_generated_im2col.cu.o.cmake:260 (message):
  Error generating file
  /home/luotao02/Paddle/build/paddle/fluid/operators/math/CMakeFiles/math_function.dir//./math_function_generated_im2col.cu.o


make[2]: *** [paddle/fluid/operators/math/CMakeFiles/math_function.dir/math_function_generated_im2col.cu.o] Error 1
make[1]: *** [paddle/fluid/operators/math/CMakeFiles/math_function.dir/all] Error 2
make: *** [all] Error 2
```
This error is similar to https://github.com/tensorflow/tensorflow/issues/10258.
The reason is that **On CUDA versions previous to 8.0, only shared variables could be declared as static in the device code.**

--- a/paddle/fluid/framework/dim.h
+++ b/paddle/fluid/framework/dim.h
@@ -158,7 +158,13 @@ HOSTDEVICE int64_t& indexer<0>(Dim<0>& dim, int idx) {
 #else
   PADDLE_ASSERT(false);
 #endif
+#if (defined __CUDA_ARCH__) && (CUDA_VERSION < 8000)
+  // On CUDA versions previous to 8.0, only __shared__ variables
+  // could be declared as static in the device code.
+  int64_t head = 0;
+#else
   static int64_t head = 0;
+#endif
   return head;
 }
 
@@ -184,7 +190,13 @@ HOSTDEVICE int64_t indexer<0>(const Dim<0>& dim, int idx) {
 #else
   PADDLE_ASSERT(false);
 #endif
+#if (defined __CUDA_ARCH__) && (CUDA_VERSION < 8000)
+  // On CUDA versions previous to 8.0, only __shared__ variables
+  // could be declared as static in the device code.
+  int64_t head = 0;
+#else
   static int64_t head = 0;
+#endif
   return head;
 }
 

